src/AppInstallerCLICore/Commands/UpgradeCommand.cpp

```diff
@@ -120,7 +120,8 @@ namespace AppInstaller::CLI
             OpenSource <<
             OpenCompositeSource(Repository::PredefinedSource::Installed);
 
-        if (context.Args.Empty())
+        if (context.Args.Empty() ||
+            (context.Args.GetArgsCount() == 1 && context.Args.Contains(Execution::Args::Type::Source)))

```

Comment:
This should be a function with a semanticly relevant name to help future proof it when new arguments are added.  For instance `ShouldListUpgrades`.